### PR TITLE
editoast: use serial_test

### LINKED ISSUE #5886: editoast(test): use 'serial_test' crate for improving test performance
### Description and goal

What's this task about, and what is it for?

Today, `editoast` tests all rely on an existing database and race conditions can happen during test execution, making the test failing. Therefore, tests are run with `--test-threads 1` to avoid these race conditions... but at the price of a longer time to run the test suite.

We could use https://crates.io/crates/serial_test to force only the needed tests to run in isolation, and keep the rest of the test suite running concurrently.

Related to https://github.com/osrd-project/osrd/issues/5441.

### Acceptance criteria

A list of conditions which define when this task shall be considered complete

The suite of tests on `editoast` run faster. If not, then we probably do not want to keep `serial_test` since it would add additional dependency and complexity for no gain.

### PR DESCRIPTION
**Pull request by Wadjetz**
close #5886 
#6200 Before
Capture d’écran 2024-01-02 à 15 51 13

#After
Screenshot 2024-01-03 at 14-59-19 editoast use serial_test · osrd-project_osrd@8ed6d06

### CODE REVIEW

File: .github/workflows/build.yml
Comment (shenriotpro) [inline .github/workflows/build.yml:1]: Should we update the editoast README as well?

**Comment by shenriotpro** [comment]
Thanks for the PR. I'm a bit skeptical for reasons I explained in https://github.com/osrd-project/osrd/issues/5886. Sorry I didn't notice earlier.

File: editoast/README.md [hunk 1]
```diff
@@ -30,12 +30,17 @@ $ curl -f http://localhost:8090/health
...
 
-To avoid thread conflicts while accessing the database, use the `--test-threads=1` option.
+To avoid thread conflicts while accessing the database, use serial_test
+
+```rust
+#[test]
+#[serial_test::serial]
+```
 
 <!--- TODO: when tests are isolated, allow multiple threads (4 may be a good tradeoff for speed/DB pool)  --->
```
Comment (Castavo) [inline editoast/README.md]: You could also remove this line

**Review by Castavo — COMMENTED** [review]
I'm also surprised that 175 tests need to be serialsed

For some of them we could do as for the rolling stocks and change the hard coded string which break unicity constraints I believe

Also, have you considered creating several serial "threads" (like described in [the docs](https://docs.rs/serial_test/2.0.0/serial_test/attr.serial.html)) ?  
I feel like not all tests which are serialized here need to be isolated from **all** other tests.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -366,7 +366,7 @@ jobs:
           docker run --name=editoast-test --net=host -v $PWD/output:/output \
             -e DATABASE_URL="postgres://osrd:password@localhost:5432/osrd" \
             ${{ fromJSON(needs.build.outputs.stable_tags).editoast-test }} \
-            /bin/bash -c "diesel migration run --locked-schema && cargo test --verbose -- --test-threads=1 && grcov . --binary-path ./target/debug/ -s . -t lcov --branch --ignore-not-existing --ignore "/*" -o /output/lcov.info"
```
Comment (shenriotpro) [inline .github/workflows/build.yml:385]: Please keep a somewhat low thread limit. In this state, tests fail on my machine (cleanup fails as well).

**Review by shenriotpro — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 2 above)
Comment (shenriotpro) [reply .github/workflows/build.yml:385]: (also applies for README)

**Review by shenriotpro — APPROVED** [review]
Thanks

**Review by Tguisnet — APPROVED** [review]
lgtm !

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**